tests/EdgeToEdge/Routes/ValidateAmountRouteTest.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Frontend\Tests\EdgeToEdge\Routes;
+
+use Symfony\Component\HttpFoundation\Request;
+use WMDE\Fundraising\Frontend\Tests\EdgeToEdge\WebRouteTestCase;
+
+class ValidateAmountRouteTest extends WebRouteTestCase {
+
+	private const PATH = '/validate-donation-amount';
+
+	/**
+	 * @dataProvider getPassingTestData
+	 */
+	public function testGivenValidParameters_successResponseIsReturned( array $parameters ): void {
+		$client = $this->createClient();
+		$client->request( Request::METHOD_POST, self::PATH, $parameters );
+		$this->assertJsonSuccessResponse( [ 'status' => 'OK' ], $client->getResponse() );
+	}
+
+	public function getPassingTestData(): array {
+		return [
+			[ [ 'amount' => '1234' ] ],
+		];
+	}
+
+	/**
+	 * @dataProvider getFailingTestData
+	 */
+	public function testGivenInvalidParameters_matchingFailureResponseIsReturned( array $parameters, array $violations ): void {
+		$client = $this->createClient();
+		$client->request( Request::METHOD_POST, self::PATH, $parameters );
+		$this->assertErrorJsonResponse( $client->getResponse() );
+		$this->assertEquals( $violations, $this->getJsonFromResponse( $client->getResponse() )['messages'] );
+	}
+
+	public function getFailingTestData(): array {
```

Comment:
AFAIK the PHPUnit naming convention is to use `failingTestDataProvider`. It certainly is what we have been doing at WMDE since we started using PHPUnit.